### PR TITLE
Add an initial project to evaluate tracking key data in governance repo.

### PR DESCRIPTION
**Pull request by brianwarner**
For discussion in the next TAC meeting, I've constructed a barebones example to start some discussions on how to make project lifecycle materials a little easier to store and find. 

I started with Enarx for no reason other than it was the first project alphabetically. In theory this could also be used for tracking status of deliverables for new project proposals, as well. Finally, another intent is to create a canonical location to store project charter documents, so every project knows where to find them should an update ever be required.

I would very much welcome and encourage thoughts on how to organize info, other info to collect, etc. All contributions are definitely welcome!

Signed-off-by: Brian Warner <brian@bdwarner.com>

### CODE REVIEW

File: TAC/Projects/Enarx/enarx-proposal.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Enarx project contribution proposal 
+
+1.  name of project
+  1. Enarx
```
Comment (thomas-fossati) [inline TAC/Projects/Enarx/enarx-proposal.md:4]: I think you need a further space here (and similarly below) to get the nesting.

**Review by thomas-fossati — COMMENTED** [review]
Thank you, LGTM -- modulo an easy to fix editorial problem (see inline).

**Review by dthaler — APPROVED** [review]

**Comment by dthaler** [comment]
Adding time bomb label to expire Feb. 15, 2022 per https://github.com/confidential-computing/governance/blob/main/tac-github-process.md

File: TAC/Projects/Enarx/enarx-proposal.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+   1.  Enarx came out of some conversations between Nathaniel McCallum and Mike Bursell (the project’s founders) and led to them presenting the idea to a number of people at Red Hat, who agreed to provide some initial resources to see if it was plausible.  When the Consortium was announced, we realised very quickly that it was very well aligned.
+
+1. statement on alignment with Consortium charter's mission
+   1.  Enarx aims to be hardware-vendor neutral, providing a run-time environment which works across platforms.  We hope to make it easy for people to deploy their own applications or applications created by 3rd party vendors, supporting the “protecting data in use” part of the charter which is core to the Consortium.  Being conceived as an open source project, we welcome input and collaboration from all parties.  Minimal Trusted Compute Base and high auditability are high on our list of design principles. The code is very modular, meaning that there may well be opportunities for other projects to re-use components for their needs. 
+
+1. link to *current* Code of Conduct (if one is adopted already)
+   1.  TBD.
```
Comment (nickvidal) [inline TAC/Projects/Enarx/enarx-proposal.md:15]: https://github.com/enarx/.github/blob/main/CODE_OF_CONDUCT.md

File: TAC/Projects/Enarx/enarx-proposal.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+1. statement on alignment with Consortium charter's mission
+   1.  Enarx aims to be hardware-vendor neutral, providing a run-time environment which works across platforms.  We hope to make it easy for people to deploy their own applications or applications created by 3rd party vendors, supporting the “protecting data in use” part of the charter which is core to the Consortium.  Being conceived as an open source project, we welcome input and collaboration from all parties.  Minimal Trusted Compute Base and high auditability are high on our list of design principles. The code is very modular, meaning that there may well be opportunities for other projects to re-use components for their needs. 
+
+1. link to *current* Code of Conduct (if one is adopted already)
+   1.  TBD. 
+
+1. sponsor from TAC, if identified (a sponsor helps mentor projects)
+   1.  Mike Bursell (Red Hat)
```
Comment (nickvidal) [inline TAC/Projects/Enarx/enarx-proposal.md:18]: Mike Bursell (Profian)

File: TAC/Projects/Enarx/enarx-proposal.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+   1.  Currently run with Nathaniel McCallum coordinating code contributions and Mike Bursell leading most other project-related matters, but both adopting an “Open Management” style with much delegation and discussion on key architectural and project matters.  We plan to revisit the governance and leadership model once we have a larger number of contributors.
+
+1. link to any documented governance practices
+   1.  n/a
+
+1. preferred maturity level (see stages below)
+   1.  Awaiting clarification on stage naming
+
+1. list of project's official communication channels (slack, irc, mailing lists)
+   1.  #enarx on freenode
```
Comment (nickvidal) [inline TAC/Projects/Enarx/enarx-proposal.md:48]: https://chat.enarx.dev/

File: TAC/Projects/Enarx/enarx-proposal.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+   1.  Issue tracking on GitHub
+
+1. link to project's website 
+   1.  https://enarx.io
```
Comment (nickvidal) [inline TAC/Projects/Enarx/enarx-proposal.md:52]: https://enarx.dev/

File: TAC/Projects/Enarx/enarx-proposal.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+1. links to social media accounts
+   1.  @enarxproject (Twitter)
+
+1. existing financial sponsorship
+   1.  Support from Red Hat
```
Comment (nickvidal) [inline TAC/Projects/Enarx/enarx-proposal.md:58]: Support from Profian

File: TAC/Projects/Enarx/enarx-proposal.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+   1.  TBD
+
+1. names of initial committers, if different from those submitting proposal
+   1.  Nathaniel McCallum, Mike Bursell
+
+1.  briefly describe the project's leadership team and decision-making process
+   1.  Currently run with Nathaniel McCallum coordinating code contributions and Mike Bursell leading most other project-related matters, but both adopting an “Open Management” style with much delegation and discussion on key architectural and project matters.  We plan to revisit the governance and leadership model once we have a larger number of contributors.
+
+1. link to any documented governance practices
+   1.  n/a
```
Comment (nickvidal) [inline TAC/Projects/Enarx/enarx-proposal.md:42]: https://github.com/confidential-computing/governance/blob/main/project-charters/Enarx_Project_Technical_Charter_2-14-2020.pdf

File: TAC/Projects/Enarx/enarx-proposal.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+   1.  Apache 2.0
+
+1. source control (GitHub by default)
+   1.  GitHub
+
+1. issue tracker (GitHub by default)
+   1.  GitHub
+
+1. external software dependencies (including licenses)
+   1.  TBD
```
Comment (nickvidal) [inline TAC/Projects/Enarx/enarx-proposal.md:30]: https://docs.google.com/spreadsheets/d/1UKnbbGWXYLjnPZsox3zmYo59nv3XSXjePfas5E2fER0/edit

File: TAC/Projects/Enarx/enarx-proposal.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+1. release methodology and mechanics
+   1.  TBD
```
Comment (nickvidal) [inline TAC/Projects/Enarx/enarx-proposal.md:33]: https://enarx.dev/docs/Contributing/Code

**Review by nickvidal — COMMENTED** [review]
Hi @brianwarner, thank you for this. I have requested some changes.

**Comment by brianwarner** [comment]
Hi @nickvidal, so this raises an interesting question. I don't have the history to know if this has already been discussed, but once a project is accepted, is the project application intended to be a living document or a static historical record? @dthaler, do you recall if this has been discussed previously?

**Comment by dthaler** [comment]
> Hi @nickvidal, so this raises an interesting question. I don't have the history to know if this has already been discussed, but once a project is accepted, is the project application intended to be a living document or a static historical record? @dthaler, do you recall if this has been discussed previously?

There's both.  We archive the version at the time of acceptance, for future reference.  But the answers to the questions are a living document as implied by the [Annual Review Process](https://github.com/confidential-computing/governance/blob/main/project-progression-policy.md#iv-annual-review-process) and so having the "current" version is important.   I think Nick is proposing this document be the latter and I think you are proposing it be the former.   Currently the former is in the groups.io file share and we don't have a "current" version so I would agree with Nick that it's more important to fill that gap.   If you want to propose moving the archived original from groups.io to here that would be ok too, but that means two copies, clearly marked in some way (directory, title of markdown, whatever).

**Comment by brianwarner** [comment]
Thanks! This is really helpful. I think the easiest approach would be to make a copy called `enarx-accepted_proposal.md` that's intended to be immutable, and rename the living one to `enarx-project_detail.md` or similar. I'll push a patch so we can see how it looks, and @nickvidal I'll accept your changes.

File: TAC/Projects/Enarx/enarx-proposal.md (hunk 2 above)
Comment (brianwarner) [reply TAC/Projects/Enarx/enarx-proposal.md]: Updated and pushed.

File: TAC/Projects/Enarx/enarx-proposal.md (hunk 4 above)
Comment (brianwarner) [reply TAC/Projects/Enarx/enarx-proposal.md]: Updated and pushed.

File: TAC/Projects/Enarx/enarx-proposal.md (hunk 6 above)
Comment (brianwarner) [reply TAC/Projects/Enarx/enarx-proposal.md]: Updated and pushed. I added Profian as a new line, not replacing Red Hat.

File: TAC/Projects/Enarx/enarx-proposal.md (hunk 3 above)
Comment (brianwarner) [reply TAC/Projects/Enarx/enarx-proposal.md]: Updated and pushed.

File: TAC/Projects/Enarx/enarx-proposal.md (hunk 5 above)
Comment (brianwarner) [reply TAC/Projects/Enarx/enarx-proposal.md]: Updated and pushed. I added this in addition to the .io domain, please let me know if that's not correct.

File: TAC/Projects/Enarx/enarx-proposal.md (hunk 7 above)
Comment (brianwarner) [reply TAC/Projects/Enarx/enarx-proposal.md]: I linked to the version in this directory. Ideally I'd suggest the charters live alongside the other docs, if that works for you?

File: TAC/Projects/Enarx/enarx-proposal.md (hunk 8 above)
Comment (brianwarner) [reply TAC/Projects/Enarx/enarx-proposal.md]: Updated and pushed.

File: TAC/Projects/Enarx/enarx-proposal.md (hunk 9 above)
Comment (brianwarner) [reply TAC/Projects/Enarx/enarx-proposal.md]: Updated and pushed.

File: TAC/Projects/README.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+## Index of Projects
+
+### Graduated
+
+* [Enarx](#enarx)
+* [Keystone](#keystone)
+* [Occlum](#occlum)
+* [Open Enclave SDK](#open-enclave-sdk)
+* [Veracruz](#veracruz)
```
Comment (dcmiddle) [inline TAC/Projects/README.md:13]: I don't think these are all graduated. Most should be Sandbox or Incubating.

**Comment by dcmiddle** [comment]
I think we can get by with one copy of each proposal rather than 'accepted' and 'detailed'. The commit history shows the initial case and then we can see each update thereafter.

**Review by dcmiddle — CHANGES_REQUESTED** [review]

**Review by thomas-fossati — APPROVED** [review]
I have reviewed the Veracruz part.  It looks good to me.  Thanks!

**Review by dcmiddle — APPROVED** [review]

**Review by dcmiddle — APPROVED** [review]

**Review by lkatalin — APPROVED** [review]

**Comment by lkatalin** [comment]
Do we want to merge this, or was there still discussion? It would be good to use the updated `TAC/Projects` link in materials I'm preparing if we think that will be the new directory structure.

**Review by thomas-fossati — APPROVED** [review]

**Pull request merged**